uint8 robotIndex
uint64 missionId